documentation/en/actor-events-api.md

```diff
@@ -0,0 +1,434 @@
+
+```ipldsch
+type ActorEvent struct {
+    entries [EventEntry] # Event entries in log form.
+    emitter Address      # Filecoin address of the actor that emitted this event.
+    # Reverted is set to true if the message that produced this event was reverted because of a
+    # network re-org in that case, the event should be considered as reverted as well.
+    reverted Bool        
+    height ChainEpoch    # Height of the tipset that contained the message that produced this event.
+    tipsetCid &Any       # CID of the tipset that contained the message that produced this event.
+    msgCid &Any          # CID of message that produced this event.
+}
+
+type EventEntry struct {
+    flags Int   # A bitmap conveying metadata or hints about this entry.
+    key String  # The key of this entry.
+    codec Int   # The value's IPLD codec.
+    value Bytes # The value of this entry as a byte string, encoded with 'codec'.
+}
+```
+
+A `flags` field is used to convey metadata or hints about the entry, currently this is used to provide an indication of the suitability of that field for indexing when stored or when used for querying.
+
+* A `flag` of `0x00` indicates that neither the key nor value are suitable for indexing.
+* A `flag` of `0x01` indicates that the key only is suitable for indexing.
+* A `flag` of `0x02` indicates that the value is suitable for indexing.
+* A `flag` of `0x03` indicates that both the key and value are suitable for indexing.
+
+Suitability for indexing is a hint, and will typically indicate the uniqueness of the content. For example if a value has an arbitrary number, such as a `piece-size`, it's unlikely to be suitable for building an index for. However, if a field such as a `"provider"` contains an actor ID, it is likely to be suitable for indexing because queries may be made for all events from a particular actor.
+
+Typically events contain entires that use either use `0x01` or `0x03` flags.
+
+The structured logging style of composition should be seen in contrast to an alternative representation as a plain map or struct where the keys represent the fields of the event and the values represent the values of those fields. Some entries may duplicate keys, in which case that particular field of the event could be represented as an array. Builtin actor events are sufficiently well defined that translation to such a format is possible, but left up to the user.
+
+## Querying Lotus for ActorEvents
+
+Two Lotus APIs are provided that can be used to obtain direct access to events stored on the node being queried (a node may not have all historical events stored and available for query):
+
+_TODO: update links_
```

Comment:
What is blocking us from updating the links?

Reply:
1. I wanted to check whether these docs are published anywhere else or if pointing into the repo is appropriate
2. These aren't in `master` yet, I'm hessitant to point a link to something that's not merged. It's on feat/nv22 and I don't know the timeframe for merging that.

Could update it now, but the links won't work till it's all landed.

Reply:
bumped to `master` even though they're not there .. they _will_ be